File: Score Counter Animated - Example Project/Score Counter Animated/Graph/ScriptTemplate/mod/CGMod.lua
local CGMod = CGMod or {}
CGMod.__index = CGMod

function CGMod.new()
    local self = setmetatable({}, CGMod)
    self.inputs = {}
    return self
end

function CGMod:setInput(index, func)
    self.inputs[index] = func
end

function CGMod:getOutput(index)
    return self.inputs[0]() % self.inputs[1]()
end

return CGMod

File: Score Counter Animated - Example Project/Score Counter Animated/Graph/ScriptTemplate/mod/CGMod.js
/**
 * @file CGMod.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGMod.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGMod extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return this.inputs[0]() % this.inputs[1]();
  }
}

exports.CGMod = CGMod;
